### DISCUSSION TITLE
Postgres compute time limit reset?

### DISCUSSION THREAD

**Discussion by Benxter83** [vercel/community#7349] [Help] [answered]
### Summary

Apparently I am hitting the postgres computing time on my hobby plan (small non-profit project for a primary school), after 3 months of development and production and as I understood properly, it doesn't get reset on the hobby plan. So the school needs to upgrade to the pro plan, which includes the first 100 hours of compute time and charges $0.1 per extra hour. 
My question now: Does the 100 hour limit of computing time get reset after 30 days (each billing cycle)? The information on the plan documentation is unclear to me.

Thank you in advance.

Benjamin Hoch

### Steps to Reproduce

N/A

**Comment by pawlean** [comment]
Hi @Benxter83!

Let me try and help you out here. 

> after 3 months of development and production and as I understood properly, it doesn't get reset on the hobby plan.

This is correct. The Hobby Plan:
- Includes a fixed amount of compute time, which is not reset every 30 days.

>  So the school needs to upgrade to the pro plan, which includes the first 100 hours of compute time and charges $0.1 per extra hour.

> Does the 100 hour limit of computing time get reset after 30 days (each billing cycle)? The information on the plan documentation is unclear to me.

This is correct. The Pro Plan:

- Includes 100 hours of compute time, which are reset every 30 days.
- Additional compute hours are billed at $0.10 per hour.
- Compute time is reset at the start of each billing cycle, giving you a fresh 100 hours.

See [relevant documentation](https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing).

Hopefully that answers your question! Let me know if anything is unclear.

_EDIT: I made an error on Hobby Plan limits. See comments below._

**Reply by Benxter83** [reply]
Hi @pawlean ! Thanks for the quick reply. 
So i read the limits for the hobby plan again and it states that also the postgres compute time are calculated on a per month basis?! 
But you are stating that it accumulates. Can you please go more into detail on this? Does that mean it must not be totaled more than 60h for 30 days or is it more like a 60h „lifetime credit“. 
You also state that if i reach the accumulated 60h i have to upgrade to pro or wait for 30 days. What happens after 30days? 
Sorry for asking so many questions. I want to keep the costs for the school to a minimum.

**Reply by pawlean** [reply]
No worries on the questions! This is all great feedback for us to improve our documentation too. Let me triple check everything again for you 😃

**Reply by pawlean** [reply, marked as answer]
@Benxter83 Totally my bad here, sorry! Just started learning about Storage on Vercel recently and made an error. For some reason I thought it was accumulated. You're right it is indeed calculated on a per-month basis, not accumulated.

- On the Hobby plan, you get 60 hours of Postgres compute time per month.
- This 60-hour limit is reset every 30 days, so you'll get a fresh 60 hours at the start of each month.
- If you exceed the 60-hour limit, you'll need to upgrade to the Pro plan or wait for the next month to start.

Does that help answer your question?

**Reply by Benxter83** [reply]
Yes that helps a lot! Thank you so much!

**Reply by pawlean** [reply]
Thanks, @Benxter83! Feel free to mark the comment as the answer as it will help other community folk asking similar questions. Let me know if you have any other questions.

**Reply by codeswitchconsulting** [reply]
@pawlean thank you for clarifying this in the post above. We have an app that has a Postgres Hobby plan and (via some automation) the compute usage of the Postgres db skyrocketed a couple of weeks ago. Could you help with these follow-up questions about the Hobby plan please...

1. Is it still valid that the 60 compute hours resets every 30 days on the Hobby plan?

2. How do we determine what the start date and end date is for this 30 day period? Is it based on a calendar month, based on when we signed up, or based on when we started using the db. For example, if we signed up and started using the db on Aug 7, would our 30 days end on Aug 31 (end of calendar month) or Sept 5 (30 days after Aug 7). From the response above, it sounds like it would end Aug 31, but just wanted to confirm.

3. If we go beyond the 60 hour compute time, are database reads/write from that point not allowed or are the hours after that 60 hour period accrued and charged to our account?

[a comment by an automated account is not shown]
